BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Contains the remote configuration for the Braintree Android SDK.
+ */
+// NEXT MAJOR VERSION: remove 'open' modifiers, Java classes/methods are open by default
+// Ref: https://kotlin-quick-reference.com/102c-R-open-final-classes.html
+open class Configuration internal constructor(configurationString: String?) {
+
+    companion object {
+        private const val ASSETS_URL_KEY = "assetsUrl"
+        private const val CLIENT_API_URL_KEY = "clientApiUrl"
+        private const val CHALLENGES_KEY = "challenges"
+        private const val ENVIRONMENT_KEY = "environment"
+        private const val MERCHANT_ID_KEY = "merchantId"
+        private const val MERCHANT_ACCOUNT_ID_KEY = "merchantAccountId"
+        private const val ANALYTICS_KEY = "analytics"
+        private const val BRAINTREE_API_KEY = "braintreeApi"
+        private const val PAYPAL_ENABLED_KEY = "paypalEnabled"
+        private const val PAYPAL_KEY = "paypal"
+        private const val KOUNT_KEY = "kount"
+        private const val GOOGLE_PAY_KEY = "androidPay"
+        private const val THREE_D_SECURE_ENABLED_KEY = "threeDSecureEnabled"
+        private const val PAY_WITH_VENMO_KEY = "payWithVenmo"
+        private const val UNIONPAY_KEY = "unionPay"
+        private const val CARD_KEY = "creditCards"
+        private const val VISA_CHECKOUT_KEY = "visaCheckout"
+        private const val GRAPHQL_KEY = "graphQL"
+        private const val SAMSUNG_PAY_KEY = "samsungPay"
+        private const val CARDINAL_AUTHENTICATION_JWT = "cardinalAuthenticationJWT"
+
+        /**
+         * Creates a new [Configuration] instance from a json string.
+         *
+         * @param configurationString The json configuration string from Braintree.
+         * @return [Configuration] instance.
+         */
+        @JvmStatic
+        @Throws(JSONException::class)
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun fromJson(configurationString: String?): Configuration {
+            return Configuration(configurationString)
+        }
+    }
+
+    private val configurationString: String
+    private val challenges: MutableSet<String> = HashSet()
+
+    private val braintreeApiConfiguration: BraintreeApiConfiguration
+    private val analyticsConfiguration: AnalyticsConfiguration
+    private val cardConfiguration: CardConfiguration
+
+    private val payPalConfiguration: PayPalConfiguration
+    private val googlePayConfiguration: GooglePayConfiguration
+
+    private val venmoConfiguration: VenmoConfiguration
+    private val kountConfiguration: KountConfiguration
+    private val unionPayConfiguration: UnionPayConfiguration
+    private val visaCheckoutConfiguration: VisaCheckoutConfiguration
+    private val graphQLConfiguration: GraphQLConfiguration
+    private val samsungPayConfiguration: SamsungPayConfiguration
```

Comment:
Is there a specific reason for the groupings here? It is not immediately clear why some of these properties have spaces and some don't.

Reply:
Could we alphabetize some of these too? Would make it easier to navigate all of the config types.